src/components/locale-selector/LocaleSelector.tsx

```diff
@@ -8,69 +7,54 @@ import SelectorModal, {
 } from "../shared/selector-modal/SelectorModal";
 import "./LocaleSelector.scss";
 
-interface ILocaleSelectorState {
-  selectorOpen: boolean;
+function buildOptions() {
+  return SUPPORTED_LOCALES.map((locale) => ({
+    key: locale,
+    text: i18n(locale),
+  }));
 }
 
-@autobind
-class LocaleSelector extends React.Component<unknown, ILocaleSelectorState> {
-  public state: ILocaleSelectorState = {
-    selectorOpen: false,
-  };
+function LocaleSelector() {
+  const [selectorOpen, setSelectorOpen] = React.useState(false);
 
-  public render() {
-    const { selectorOpen } = this.state;
-
-    return (
-      <>
-        <IconButton
-          className="locale-selector__button"
-          iconName="translate"
-          text={i18n("change_language")}
-          onClick={this.openSelector}
-        />
-
-        <SelectorModal
-          className="locale-selector__modal"
-          title={i18n("change_language")}
-          closeButton={true}
-          onOptionSelected={this.onOptionSelected}
-          options={this.buildOptions()}
-          open={selectorOpen}
-          onClose={this.closeSelector}
-        />
-      </>
-    );
-  }
-
-  private buildOptions() {
-    return SUPPORTED_LOCALES.map((locale) => ({
-      key: locale,
-      text: i18n(locale),
-    }));
+  function openSelector() {
+    setSelectorOpen(true);
   }
 
-  private closeSelector() {
-    this.setState({
-      selectorOpen: false,
-    });
+  function closeSelector() {
+    setSelectorOpen(false);
   }
 
-  private openSelector() {
-    this.setState({
-      selectorOpen: true,
-    });
-  }
-
-  private onOptionSelected(option: ISelectorModalOption) {
+  function onOptionSelected(option: ISelectorModalOption) {
     const lang = option.key;
 
     AppSettings.settings.locale = lang;
     AppSettings.save();
     Locale.setLocale(lang);
 
-    this.closeSelector();
+    closeSelector();
   }
+
+  return (
+    <>
+      <IconButton
+        className="locale-selector__button"
+        iconName="translate"
+        text={i18n("change_language")}
+        onClick={openSelector}
+      />
+
+      <SelectorModal
+        className="locale-selector__modal"
+        title={i18n("change_language")}
+        closeButton={true}
+        onOptionSelected={onOptionSelected}
+        options={buildOptions()}
```

Comment:
Yeah I thought the same thing while refactoring, just left it like it was since it didn't really makes a difference in the end and it was like this before, but i agree and can change it 🙂